src/ucp/core/ucp_worker.c

```diff
@@ -2526,18 +2528,36 @@ static void ucp_worker_destroy_eps(ucp_worker_h worker,
     }
 }
 
+static void ucp_worker_eps_cleanup(ucp_worker_h worker)
+{
+    /* Cleanup hash of discarded UCT EPs prior destroying all UCP EPs to
+     * destroy UCP EPs which were already marked as closed by a user, but had
+     * discarding is in progress */
+    ucp_worker_discard_uct_ep_cleanup(worker);
+    ucp_worker_destroy_eps(worker, &worker->all_eps, "all");
+    ucp_worker_destroy_eps(worker, &worker->internal_eps, "internal");
+    /* Cleanup hash of discarded UCT EPs one more time to stop discarding of
+     * UCT EPs (it could be discarding of AUX EPs from WIREUP EPs) which might
+     * be started during destroying of UCP EPs */
+    ucp_worker_discard_uct_ep_cleanup(worker);
```

Comment:
i'd suggest trying to find another way to fix it, now it looks like happy coincidence that it fixes the problem, because we manipulating by the refcounter in unclear way

Reply:
will think about it